File: quickshop/app/src/main/java/com/avas/quickshop/ui/theme/app_config/AppConfig.kt
package com.avas.quickshop.ui.theme.app_config


/// Created by Aabhash Shakya on 18/11/2025

import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json

@Serializable
data class AppConfig(val uuid: String, val shouldCacheUUID: Boolean, val theme: AppTheme) {
    fun toJson(): String = Json.encodeToString(this)
}

@Serializable
data class AppTheme(
    val themeMode: String,
    val colors: AppColorsPair,
    val spacing: AppSpacingConfig,
    val typography: AppTypographyConfig
)

@Serializable
data class AppColors(
    val primary: String,
    val secondary: String,
    val surface: String,
    val onSurface: String,
    val onPrimary: String,
    val onSecondary: String,
    val primaryContainer: String,
    val onPrimaryContainer: String
)

@Serializable
data class AppColorsPair(val light: AppColors, val dark: AppColors)

@Serializable
data class AppSpacingConfig(val small: Int, val medium: Int, val large: Int)

@Serializable
data class AppTypographyConfig(
    val titleLargeSize: Int,
    val bodyLargeSize: Int,
    val labelSmallSize: Int,
    val titleFont: String,
    val bodyFont: String,
    val labelFont: String
)

File: QuickShop-Ios/QuickShop-Ios/flutter_utils/AppConfig.swift
//
//  AppConfig.swift
//  quickshop-ios
//
//  Created by Aabhash Shakya on 21/11/2025.
//

import Foundation

// MARK: - AppConfig (Root)
struct AppConfig: Codable {
    let uuid: String
    let shouldCacheUUID: Bool
    let theme: AppTheme
}

// MARK: - Theme
struct AppTheme: Codable {
    let themeMode: String
    let colors: AppColorsPair
    let spacing: AppSpacingConfig
    let typography: AppTypographyConfig
}

// MARK: - Colors + Pair
struct AppColorsPair: Codable {
    let light: AppColors
    let dark: AppColors
}

struct AppColors: Codable {
    let primary: String
    let secondary: String
    let surface: String
    let onSurface: String
    let onPrimary: String
    let onSecondary: String
    let primaryContainer: String
    let onPrimaryContainer: String
}

// MARK: - Spacing
struct AppSpacingConfig: Codable {
    let small: Int
    let medium: Int
    let large: Int
}

// MARK: - Typography
struct AppTypographyConfig: Codable {
    let titleLargeSize: Int
    let bodyLargeSize: Int
    let labelSmallSize: Int
    let titleFont: String
    let bodyFont: String
    let labelFont: String
}
